src/main/java/com/selfxdsd/todos/Puzzle.java

```diff
@@ -357,6 +382,18 @@ public Puzzle build() {
                 throw new IllegalStateException("Time is missing");
             }
 
+            final String id = this.id;
```

Comment:
@amihaiemil Here is fixed a bug triggered when builder is reused to create puzzles in a loop and then add them to a list. The created `Puzzle` kept reference to the latest vals of the mutable fields of the builder.
This means that all the puzzles in the list will have the latest builder values.
By scoping the fields to locals this behaviour is avoided.